### PR TITLE
add color extension to support creating colors from hex value

### PR DESCRIPTION
**Pull request by mormo17**

### CODE REVIEW

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//  Created by Mariam Ormotsadze on 04.05.23.
+//
+
+import SwiftUI
+import UIKit
+
+public extension Color {
+    init?(hex: String) {
+        var hexSanitized = hex.trimmingCharacters(in: .whitespacesAndNewlines)
+        hexSanitized = hexSanitized.replacingOccurrences(of: "#", with: "")
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:14]: Use function calling chain and change var to let

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        var rgb: UInt64 = 0
+
+        var r: CGFloat = 0.0
+        var g: CGFloat = 0.0
+        var b: CGFloat = 0.0
+        var a: CGFloat = 1.0
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:21]: Do not use default values and use let

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            r = CGFloat((rgb & 0xFF000000) >> 24) / 255.0
+            g = CGFloat((rgb & 0x00FF0000) >> 16) / 255.0
+            b = CGFloat((rgb & 0x0000FF00) >> 8) / 255.0
+            a = CGFloat(rgb & 0x000000FF) / 255.0
+
+        } else {
+            return nil
+        }
+
+        self.init(red: r, green: g, blue: b, opacity: a)
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:42]: Changes names to red blue...

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+}
+
+public extension Color {
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:46]: Why new extension, above there was an extension scope

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    func toHex() -> String? {
+        let uic = UIColor(self)
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:48]: uiColor

File: Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        } else {
+            return nil
+        }
+
+        self.init(red: r, green: g, blue: b, opacity: a)
+    }
+}
+
+public extension Color {
+    func toHex() -> String? {
```
Comment (Khizanag) [inline Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:47]: Why do we need this one?

Comment (mormo17) [reply Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:47]: We don't

Comment (Khizanag) [reply Chamasma/Common/Sources/Common/Extensions/Colors+Extension.swift:47]: Then remove it please

**Pull request merged**